src/NATS.Client.JetStream/JSContext.cs

```diff
@@ -15,23 +21,158 @@ public JSContext(NatsConnection nats, JSOptions options)
         _options = options;
     }
 
-    public async ValueTask<JSStream> CreateStream(Action<StreamCreateRequest> request)
+    public ValueTask<JSResponse<StreamInfo>> CreateStreamAsync(
+        StreamConfiguration request,
+        CancellationToken cancellationToken = default) =>
+        JSRequestAsync<StreamConfiguration, StreamInfo>(
+            subject: $"{_options.Prefix}.STREAM.CREATE.{request.Name}",
+            request,
+            cancellationToken);
+
+    public ValueTask<JSResponse<StreamMsgDeleteResponse>> DeleteStreamAsync(
+        string stream,
+        CancellationToken cancellationToken = default) =>
+        JSRequestAsync<object, StreamMsgDeleteResponse>(
+            subject: $"{_options.Prefix}.STREAM.DELETE.{stream}",
+            null,
+            cancellationToken);
+
+    public ValueTask<JSResponse<ConsumerInfo>> CreateConsumerAsync(
+        ConsumerCreateRequest request,
+        CancellationToken cancellationToken = default) =>
+        JSRequestAsync<ConsumerCreateRequest, ConsumerInfo>(
+            subject: $"{_options.Prefix}.CONSUMER.CREATE.{request.StreamName}.{request.Config.Name}",
+            request,
+            cancellationToken);
+
+    public async ValueTask<PubAckResponse> PublishAsync<T>(
+        string subject,
+        T? data,
+        NatsPubOpts opts = default,
+        CancellationToken cancellationToken = default)
+    {
+        await using var sub = await _nats.RequestSubAsync<T, PubAckResponse>(
+                subject: subject,
+                data: data,
+                requestOpts: opts,
+                replyOpts: default,
+                cancellationToken)
+            .ConfigureAwait(false);
+
+        if (await sub.Msgs.WaitToReadAsync(CancellationToken.None).ConfigureAwait(false))
```

Comment:
Saw `CancellationToken.None` in 3 different places - should it be passing `cancellationToken` or is it intentional to use a different cancellation token for these particular calls?

Reply:
It is intentional because the token passed to the subscription should complete the channel to allow graceful exit from the function. That's the idea anyway.